From a49e6ad5c3a4f337928b590e1a004e12c501ed75 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Mon, 27 Mar 2023 22:27:29 +0000
Subject: [PATCH] CMakeLists.txt: replace -std=c++0x with -std=c++14

:Release Notes:
c++0x isn't compatible with googletest currently in mickledore.

:Detailed Notes:
With googletest 1.13.0 from:
https://git.openembedded.org/meta-openembedded/commit/?id=46266e59ff36b38f46a67573528cee658960c764
it fails to build with:
http://gecko.lge.com:8000/Errors/Details/561645

FAILED: test/CMakeFiles/all-tests.dir/test_whiteout.cpp.o
TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot-native/usr/bin/x86_64-webos-linux/x86_64-webos-linux-g++  -ITOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/git/include -isystem TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot/usr/src/gtest/include -isystem TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot/usr/src/gtest -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -Werror=return-type  --sysroot=TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/git=/usr/src/debug/leveldb-tl/0.1.6-r4  -fdebug-prefix-map=TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/git=/usr/src/debug/leveldb-tl/0.1.6-r4  -fmacro-prefix-map=TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/build=/usr/src/debug/leveldb-tl/0.1.6-r4  -fdebug-prefix-map=TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/build=/usr/src/debug/leveldb-tl/0.1.6-r4  -fdebug-prefix-map=TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot=  -fmacro-prefix-map=TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot=  -fdebug-prefix-map=TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot-native=  -fvisibility-inlines-hidden  -std=c++0x -Wall -Wextra -Wconversion -Werror -pthread -MD -MT test/CMakeFiles/all-tests.dir/test_whiteout.cpp.o -MF test/CMakeFiles/all-tests.dir/test_whiteout.cpp.o.d -o test/CMakeFiles/all-tests.dir/test_whiteout.cpp.o -c TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/git/test/test_whiteout.cpp
In file included from TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot/usr/include/gtest/gtest-message.h:57,
                 from TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot/usr/include/gtest/gtest-assertion-result.h:46,
                 from TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot/usr/include/gtest/gtest.h:64,
                 from TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/git/test/test_whiteout.cpp:23:
TOPDIR/BUILD/work/qemux86_64-webos-linux/leveldb-tl/0.1.6-r4/recipe-sysroot/usr/include/gtest/internal/gtest-port.h:270:2: error: #error C++ versions less than C++14 are not supported.
  270 | #error C++ versions less than C++14 are not supported.
      |  ^~~~~

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-412] Create GPVB with Yocto 4.2 Mickledore

Change-Id: I19e28f7ab38ecfbfe248d657506e237e8b648a5b
---
Upstream-Status: Pending

 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 73302c5..6a0e81f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,7 @@ if (BUILD_TESTING)
     find_package(GTest REQUIRED)
 endif()
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wconversion -Werror")
 
 include_directories(include)
